### PR TITLE
cleanup: get stream info simply

### PR DESCRIPTION
**Pull request by mrkm4ntr**
Signed-off-by: mrkm4ntr <mrkm4ntr@gmail.com>

Commit Message: Get stream info from the null checked downstream variable.
Additional Description:
Risk Level: low
Testing: n/a
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Comment by wbpcode** [comment]
Thanks for the cleanup. But please fix he format error by running the tool "./ci/do_ci.sh fix_format"

File: source/common/tcp_proxy/tcp_proxy.cc [hunk 1]
```diff
@@ -362,12 +362,11 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
   }
 
   if (auto downstream_connection = downstreamConnection(); downstream_connection != nullptr) {
-    if (!read_callbacks_->connection()
-             .streamInfo()
+    if (!downstream_connection->streamInfo()
```
Comment (soulxu) [inline source/common/tcp_proxy/tcp_proxy.cc:365]: nit, seems this can be `getStreamInfo` also.

File: source/common/tcp_proxy/tcp_proxy.cc [hunk 2]
```diff
@@ -362,12 +362,11 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
...
              .filterState()
-             ->hasData<Network::ProxyProtocolFilterState>(
+             .hasData<Network::ProxyProtocolFilterState>(
                  Network::ProxyProtocolFilterState::key())) {
-      read_callbacks_->connection().streamInfo().filterState()->setData(
+      getStreamInfo().filterState()->setData(
```
Comment (wbpcode) [inline source/common/tcp_proxy/tcp_proxy.cc:369]: `downstream_connection->streamInfo()` may be better here to keep consistent with the `if` check. 🤔

**Comment by wbpcode** [comment]
Thanks for the cleanup and a minor comment is added. 🌷 

ps: please do not using `forsh push`, it may cause some problems for reviewers, thanks.

File: source/common/tcp_proxy/tcp_proxy.cc (hunk 1 above)
Comment (wbpcode) [reply source/common/tcp_proxy/tcp_proxy.cc:365]: `getStreamInfo` will introduces some unnecessary calls `read_callbacks_->connection()` (although this overhead is almost negligible).

File: source/common/tcp_proxy/tcp_proxy.cc (hunk 2 above)
Comment (mrkm4ntr) [reply source/common/tcp_proxy/tcp_proxy.cc:369]: `downstream_connection` is a const value, so we cannot invoke setData of stream info from the `downstream_connection`. The stream info from getStreamInfo is not const.

Comment (wbpcode) [reply source/common/tcp_proxy/tcp_proxy.cc:369]: I am not sure. 🤔 But can we just simplify the codes to something like this:

```
  auto& downstream_connection = read_callbacks_->connection();
  auto& filter_state = downstream_connection.streamInfo().filterState();
  if (!filter_state->hasData<Network::ProxyProtocolFilterState>(
          Network::ProxyProtocolFilterState::key())) {
    filter_state->setData(
        Network::ProxyProtocolFilterState::key(),
        std::make_shared<Network::ProxyProtocolFilterState>(Network::ProxyProtocolData{
            downstream_connection.connectionInfoProvider().remoteAddress(),
            downstream_connection.connectionInfoProvider().localAddress()}),
        StreamInfo::FilterState::StateType::ReadOnly,
        StreamInfo::FilterState::LifeSpan::Connection);
  }
  transport_socket_options_ =
      Network::TransportSocketOptionsUtility::fromFilterState(*filter_state);

  if (auto typed_state = filter_state->getDataReadOnly<Network::UpstreamSocketOptionsFilterState>(
          Network::UpstreamSocketOptionsFilterState::key());
      typed_state != nullptr) {
    auto downstream_options = typed_state->value();
    if (!upstream_options_) {
      upstream_options_ = std::make_shared<Network::Socket::Options>();
    }
    Network::Socket::appendOptions(upstream_options_, downstream_options);
  }
```

Comment (mrkm4ntr) [reply source/common/tcp_proxy/tcp_proxy.cc]: Ah, got it. No need to use `downstreamConnection()`. Thanks.

File: source/common/tcp_proxy/tcp_proxy.cc [hunk 3]
```diff
@@ -362,34 +362,28 @@ Network::FilterStatus Filter::initializeUpstreamConnection() {
     return Network::FilterStatus::StopIteration;
   }
 
-  if (auto downstream_connection = downstreamConnection(); downstream_connection != nullptr) {
```
Comment (antoniovicente) [inline source/common/tcp_proxy/tcp_proxy.cc:365]: The removal of this nullptr check worried me, but if all tests pass and there are no comments explaining why this nullptr check is needed I have to guess that this change is fine.

**Review by antoniovicente — APPROVED** [review]
thanks for the code cleanup.

**Comment by rojkov** [comment]
@yanavlasov gentle ping

**Review by yanavlasov — APPROVED** [review]

**Pull request merged**